### PR TITLE
[Feature] Subscribe & publish_value API kuksa.val.v2

### PR DESCRIPTION
**Pull request by lukasmittag**
This adds the features of kuksa.val.v2 to subscribe and publish a value to databroker. It fixes some conversion functions as well. It adds a unit test for subscribe function that leverages `publish_value`.

### CODE REVIEW

File: databroker/src/grpc/kuksa_val_v2/val.rs [hunk 1]
```diff
@@ -65,37 +77,122 @@ impl proto::val_server::Val for broker::DataBroker {
...
+        let request = request.into_inner();
+
+        let mut updates: HashMap<i32, proto::Datapoint> = HashMap::new();
+
+        updates.insert(
+            get_signal_id(request.signal_id, &broker).await.unwrap(),
+            request.data_point.unwrap(),
+        );
+
+        let values_request: proto::PublishValuesRequest = proto::PublishValuesRequest {
```
[the remark this answers is not in the document]
Comment (lukasmittag) [reply databroker/src/grpc/kuksa_val_v2/val.rs:254]: but then I do not need a separate function. which is totally fine for me. Just thought reusing makes sense.

File: databroker/src/grpc/kuksa_val_v2/val.rs [hunk 2]
```diff
@@ -440,6 +640,337 @@ mod tests {
...
+            Ok(response) => {
+                // Handle the successful response
+                let publish_response = response.into_inner();
+
+                // Check if there is an error in the response
+                if let Some(error) = publish_response.error {
+                    assert!(f, "Publish failed with error: {:?}", error);
+                } else {
+                    println!("Publish succeeded.");
+                }
```
Comment (rafaeling) [inline databroker/src/grpc/kuksa_val_v2/val.rs:733]: Here you just want to assert that the error is `None` right? You can not assert both with same unit test.
```suggestion
                assert_eq!(publish_response.error, None);
```

File: databroker/src/grpc/kuksa_val_v2/val.rs [hunk 3]
```diff
@@ -440,6 +640,337 @@ mod tests {
...
```
Comment (rafaeling) [inline databroker/src/grpc/kuksa_val_v2/val.rs:880]: Same here
```suggestion
                assert_eq!(publish_response.error, None);
```

File: databroker/src/grpc/kuksa_val_v2/val.rs [hunk 4]
```diff
@@ -440,6 +640,337 @@ mod tests {
...
```
Comment (rafaeling) [inline databroker/src/grpc/kuksa_val_v2/val.rs:912]: Same here
```suggestion
                assert_eq!(publish_response.error, None);
```

**Comment by rafaeling** [comment]
Looks good to me, just few small changes.

**Review by rafaeling — APPROVED** [review]
Looks good to me

**Pull request merged**